File: magix_api/Models/Deck.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace magix_api;

public partial class Deck
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public int PlayerId { get; set; }
    public virtual Player? Player { get; set; }
    public required string Name { get; set; }
    public int HeroId { get; set; }
    public virtual Hero? Hero { get; set; }
    public int TalentId { get; set; }
    public virtual Talent? Talent { get; set; }
    public int FactionId { get; set; }
    public virtual Faction? Faction { get; set; }
    public bool Active { get; set; } = false;
    public virtual ICollection<Game>? Games { get; }
    public virtual ICollection<DeckCard> DeckCards { get; set; } = new List<DeckCard>();
    [NotMapped]
    public List<Card>? Cards { get; set; }
}

File: frontend/src/types/Deck/Deck.ts
import Card from "@customTypes/Card";
import Faction from "@customTypes/Faction";
import Hero from "@customTypes/Hero";
import Talent from "@customTypes/Talent";

type Deck = {
	id?: string;
	name: string;
	hero: Hero;
	talent: Talent;
	faction: Faction;
	active: boolean;
	cards: Card[];
};
export default Deck;
